cocotb/clock.py

```diff
@@ -65,12 +65,51 @@ class Clock(BaseClock):
         signal: The clock pin/signal to be driven.
         period (int): The clock period. Must convert to an even number of
             timesteps.
-        units (str, optional): One of 
+        units (str, optional): One of
             ``None``, ``'fs'``, ``'ps'``, ``'ns'``, ``'us'``, ``'ms'``, ``'sec'``.
             When no *units* is given (``None``) the timestep is determined by
             the simulator.
+
+
+    If you need more features like a phase shift and an asymmetric duty cycle,
+    it is simple to create your own clock generator (that you then :func:`fork`):
+
+    .. code-block:: python
+
+        @cocotb.coroutine
+        def custom_clock():
+            # pre-construct triggers for performance
+            high_time = Timer(high_delay, units="ns")
+            low_time = Timer(low_delay, units="ns")
+            yield Timer(initial_delay, units="ns")
+            while True:
+                dut.clk <= 1
+                yield high_time
+                dut.clk <= 0
+                yield low_time
+
+    If you also want to change the timing during simulation,
+    use this slightly more inefficient example instead where 
+    the :class:`Timer`\ s inside the while loop are created with 
```

Comment:
The flake8 check in CI fails here: 
`./cocotb/clock.py:93:23: W605 invalid escape sequence '\ '` (https://travis-ci.org/cocotb/cocotb/jobs/612518805?utm_medium=notification&utm_source=github_status)

Reply:
Needs either a double backslash, or an `r` on the docstring

Reply:
Fixed it myself